tools/iree-e2e-matmul-test.cc

```diff
@@ -0,0 +1,1247 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/cpu.h"
+#include "iree/base/internal/flags.h"
+#include "iree/base/internal/math.h"
+#include "iree/base/internal/path.h"
+#include "iree/hal/api.h"
+#include "iree/modules/hal/module.h"
+#include "iree/tooling/context_util.h"
+#include "iree/tooling/device_util.h"
+#include "iree/tooling/vm_util.h"
+#include "iree/vm/api.h"
+#include "iree/vm/native_module_cc.h"
+
+IREE_FLAG(bool, require_exact_results, true,
+          "Requires floating point result elements to match exactly.");
+IREE_FLAG(
+    float, acceptable_fp_delta, 1e-5f,
+    "Maximum absolute difference allowed with inexact floating point results.");
```

Comment:
Do we have a similar fuzzy match for `iree-run-module --expected_output`? I'm wondering about finally removing the 'check' module in favor of more logic in IR (I have a few other ideas for test suite refactorings, fuzzy comparison is one of the baseline requirements for test tools).

https://github.com/openxla/iree/blob/d71c14708be2b0ae07e8567b4737b0803f4acee9/runtime/src/iree/tooling/run_module.c#L68-L73

`vm.check.nearly_eq` (https://github.com/openxla/iree/issues/5854) is also related

Reply:
We do in comparison.h/comparison.cc.